specs/http-api/README.md

```diff
@@ -323,18 +367,39 @@ Uses DID authn via Bearer token in header.
 ## List Exchanges
 
 ### Description
-Returns an array of tbdex message arrays
-
-### Authentication
-Uses DID authn via Bearer token in header.
+Returns an array of tbdex message arrays (a list of exchanges)
 
 ### Endpoint
 `GET /exchanges`
 
+### Authentication
+A tbdex http client must send an Http Auth Header in the form of a JWT `Bearer` token.
+
+The JWT must consist of JWS header and JWT payload containing the below fields, and signed by the client's DID.
+
+#### JWS Header
+| Param  | Value | Description | 
+|--------|-------|-------------|
+| `typ`  | `JWT` | Token type  |
+| `alg`  | `HS256` | Algorithm |
+| `kid`  | `did:dht:alice...#dwn-sig` | Key ID of the sender's DID |
+
+#### JWS Payload
+| Param  | Value | Description | 
+|--------|-------|-------------|
+| `aud`  | `did:dht:pfi` | DID of the audience (i.e. PFI DID)  |
+| `iss`  | `did:dht:alice` | DID of the issuer (i.e. Alice DID) |
+| `exp`  | `1703222687` | Time when the JWT expires. 1 minute after issuance  |
+| `iat`  | `1703222795` | Time when the JWT was issued. |
+| `jti`  |  `0189f7e5-c883-7106-8272-ccb7fcba0575` | Unique identifier for the JWT. Used to prevent reply attacks. Must be uuidv7. |
```

Comment:
> It's unclear to me how this would prevent replay attacks

@mistermoe and i chatted about this briefly. If the PFI wanted to prevent replay attacks, they'd want to keep the `jti` fields of each request in a cache / log, and check whether the same `jti` fields are sent by subsequent requests. If a token with same `jti` is sent again, then this could be an attacker using the same token to perform actions not intended by the original token sender.

To reduce the amount of `jti` the PFI would have to store at any given time, we also have `exp` field in the claimsset, which we've tentatively set to 1 min from current time. A PFI could implement a TTL cache so that any expired token falls off the cache automatically.

> Who is responsible for generating this nonce?

We would apply this spec to building tbdex http-client library, specifically, in the implementation of `TbdexHttpClient.generateRequestToken()` method, which generates a Bearer JWT token which is then used to make a call to, say, `TbdexHttpClient.getExchanges()`.

> How does the verifier (i.e. the PFI) know/trust the method used for the nonce generation?

In tbdex http-client, we were thinking of using an open source uuidv7 generation library. I was looking at [this one](https://github.com/f4b6a3/uuid-creator?tab=readme-ov-file) for tbdex http-client in kotlin.

Reply:
Thanks for the details!

It would appear to me that an attacker can still replay the message by waiting long enough for the cache to evict the token. Am I understanding this correctly?

Without understanding the details, I think this is being used to guarantee idempotency. A side effect of idempotency on an endpoint is that replay attacks become meaningless. Is the motivation here to address idempotency of the api, to prevent replay attacks, or to solve both?

Reply:
> an attacker can still replay the message by waiting long enough for the cache to evict the token

they could, but during the `verifyRequestToken()` stage after the PFI receives the token for an endpoint, the cache check would not prevent the replay since it will have fallen off the cache, but an expiry check would still prevent the replay. i wrote a bit more about what verifyRequestToken does [here](https://github.com/TBD54566975/tbdex/pull/216/files#diff-1ab7678648ffe33c3535373210080ab41cd2c38573890fdfcf008700ef14c3c7R215-R218)

the motivation here is to solve for both idempotency and replay attack prevention